indexer/client_auth.py

```diff
@@ -0,0 +1,125 @@
+    oauth_group.add_argument('--oauth-client-secret',
+                             default=config['OAUTH_CLIENT_SECRET'],
+                             help='The OAuth client secret, to enable client credential mode')
+
+
+def create_file_catalog_rest_client(args: Namespace) -> RestClient:
+    """Create a RestClient from argparse args."""
+    if args.oauth_client_secret:
+        LOG.debug('Using client credentials to authenticate with the File Catalog')
+        return ClientCredentialsAuth(
+            address=args.file_catalog_rest_url,
+            token_url=args.oauth_url,
+            client_id=args.oauth_client_id,
+            client_secret=args.oauth_client_secret,
+            timeout=args.rest_timeout,
+            retries=args.rest_retries,
+        )
+    else:
+        # raise an error if we can't honor the user's request
+        if args.rest_retries != defaults.REST_RETRIES:
+            raise ValueError("user credentials do not support custom REST_RETRIES or --rest-retries")
+        if args.rest_timeout != defaults.REST_TIMEOUT:
+            raise ValueError("user credentials do not support custom REST_TIMEOUT or --rest-timeout")
+
+        # otherwise, let's make that RestClient
+        LOG.debug('Using user credentials to authenticate with the File Catalog')
+        if args.oauth_client_id == 'file-catalog-indexer':
+            args.oauth_client_id = 'file-catalog-indexer-public'
+        return SavedDeviceGrantAuth(
+            address=args.file_catalog_rest_url,
+            filename='.file-catalog-auth',
+            token_url=args.oauth_url,
+            client_id=args.oauth_client_id,
+        )
+
+
+def create_iceprod_rest_client(args: Namespace) -> RestClient:
+    """Create a RestClient from argparse args."""
+    if args.oauth_client_secret:
+        LOG.debug('Using client credentials to authenticate with IceProd')
+        return ClientCredentialsAuth(
+            address=args.iceprod_rest_url,
+            token_url=args.oauth_url,
+            client_id=args.oauth_client_id,
+            client_secret=args.oauth_client_secret,
+            timeout=args.rest_timeout,
+            retries=args.rest_retries,
+        )
+    else:
+        # raise an error if we can't honor the user's request
+        if args.rest_retries != defaults.REST_RETRIES:
+            raise ValueError("user credentials do not support custom REST_RETRIES or --rest-retries")
+        if args.rest_timeout != defaults.REST_TIMEOUT:
+            raise ValueError("user credentials do not support custom REST_TIMEOUT or --rest-timeout")
+
+        # otherwise, let's make that RestClient
+        LOG.debug('Using user credentials to authenticate with IceProd')
+        if args.oauth_client_id == 'file-catalog-indexer':
+            args.oauth_client_id = 'file-catalog-indexer-public'
+        return SavedDeviceGrantAuth(
+            address=args.iceprod_rest_url,
+            filename='.iceprod-auth',
```

Comment:
I think you want this to use the same file as the file catalog auth, or you'll need to log in a second time.  If you're ok logging in a second time, then the client_id should be `iceprod-public` to separate things out.